core/types/src/report.rs

```diff
@@ -126,6 +128,75 @@ new_type_accessors_impls! {
     ReportBody, sgx_report_body_t;
 }
 
+impl From<[u8; mem::size_of::<sgx_report_body_t>()]> for ReportBody {
+    fn from(bytes: [u8; mem::size_of::<sgx_report_body_t>()]) -> Self {
+        // A note about the `expect()` calls.  This size is specified in the
+        // signature and there are unit tests ensuring the extraction from a
+        // byte array.  The values should not fail to extract due to size
+        // issues.
+        let mut body = Self::default();
+
+        let cpu_svn: [u8; SGX_CPUSVN_SIZE] =
```

Comment:
This could use mem transmute or similar, not sure if we have a preference

Reply:
We should prefer safe code to unsafe ;-)